source/python/idetect/model.py

```diff
@@ -65,35 +65,76 @@ class Article(Base):
     publication_date = Column(DateTime)
     retrieval_date = Column(DateTime)
     created = Column(DateTime(timezone=True), server_default=func.now())
-    updated = Column(DateTime(timezone=True), onupdate=func.now())
+    updated = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
     content = relationship(
         'Content', secondary=article_content, back_populates='article')
     reports = relationship('Report')
 
-    def update_status(self, new_status):
+    def __str__(self):
+        return "<Article {} {} {}>".format(self.id, self.url_id, self.url)
+
+    def get_updated_version(self):
+        """Return the most recent version of this article by updated timestamp"""
+        return Article.get_latest_version(object_session(self), self.url_id)
+
+    @classmethod
+    def get_latest_version(cls, session, url_id=None, url=None, lock=False):
+        """
+        Return the Article with the given url_id or url with the most recent updated timestamp.
+        If lock is True, uses FOR UPDATE locking, so that other calls to get_latest_version will
+        block until the transaction is committed or rolled back.
         """
-        Atomically Update the status of this Article from to new_status.
-        If something changed the status of this article since it was loaded, raise.
+        query = session.query(cls)
+        if url_id is not None:
+            query = query.filter(Article.url_id == url_id)
+        if url is not None:
+            query = query.filter(Article.url == url)
+        # if neither url_id nor url is provided, this returns the most recent article overall
```

Comment:
There probably isn't, in practice, but if both are there only a row that matches both will be returned.